projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -218,7 +218,7 @@ static function ( $item, $old_slug ) {
 				isset( $submenu_item[1] ) ? $submenu_item[1] : 'read',
 				$submenus_to_update[ $submenu_item[2] ],
 				'',
-				$i
+				0 === $i ? 0 : $i + 1
```

Comment:
Internally, add_submenu_page() function has two code paths that we are interested in:
- when the position is 0, the item is appended to the submenu array
- when the position is non-0, the existing submenu array will be split in two arrays using `array_slice( $submenu[ $parent_slug ], 0, $position, true );` call. The third parameter is in fact the length of the array, therefore, we need to increment with one the actual position where we should place the newly created element. However, when we enter the first code path (append), we should still point the position to 0 (as it was before).